### PR TITLE
feat(addons): add maintenance windows manipulation with the new `addons-config` command

### LINKED ISSUE #952: [cli] Maintenance windows should be visible/editable

### PR DESCRIPTION
**Pull request by SCedricThomas**
fix #952

- [x] Add a changelog entry in the section "To Be Released" of CHANGELOG.md

### CODE REVIEW

**Comment by SCedricThomas** [comment]
We will need a new release of go-scalingo in order to pass the CI but can you take a look at my code ? 
You can take a look to the specifications [here](https://github.com/Scalingo/project-items/issues/283).
Here is my output: 
image

File: cmd/addons.go [hunk 1]
```diff
@@ -160,4 +160,46 @@ var (
...
+		Category: "Addons",
+		Usage:    "Configure an add-on attached to your app",
+		Flags: []cli.Flag{&appFlag,
+			&cli.StringFlag{Name: "maintenance-window-hour", Usage: "Configure add-on maintenance window starting hour"},
+			&cli.StringFlag{Name: "maintenance-window-day", Usage: "Configure add-on maintenance window day"},
+		},
+		ArgsUsage: "addon-id",
+		Description: CommandDescription{
+			Description: "Configure an add-on attached to your app",
+			Examples:    []string{"scalingo --app manifest addons-config --maintenance-window-day tuesday --maintenance-window-hour 2 addon_uuid"},
```
Comment (EtienneM) [inline cmd/addons.go:174]: suggestion: It should not only take an addon UUID. It should be possible to provide the database type (e.g. `postgresql`). Cf. what is done in the `logs` command: https://github.com/Scalingo/cli/blob/6ef569ab159e38e69ef8372d6e287dd197796916/db/logs.go#L28-L35

consistency: example app name is _always_ `my-app`. Not `manifest`.

File: addons/config.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		strings.ToLower(time.Tuesday.String()):   time.Tuesday,
+		strings.ToLower(time.Wednesday.String()): time.Wednesday,
+		strings.ToLower(time.Thursday.String()):  time.Thursday,
+		strings.ToLower(time.Friday.String()):    time.Friday,
+		strings.ToLower(time.Saturday.String()):  time.Saturday,
+	}
+
+	c, err := config.ScalingoClient(ctx)
+	if err != nil {
+		return errgo.Notef(err, "fail to get Scalingo client")
```
Comment (EtienneM) [inline addons/config.go:34]:
```suggestion
		return errgo.Notef(err, "get Scalingo client to update addon config")
```

**Review by EtienneM — CHANGES_REQUESTED** [review]
The failed CI and the release of go-scalingo are 2 completely different things. I don't understand why there is a link between these two?

File: cmd/addons.go (hunk 1 above)
Comment (SCedricThomas) [reply cmd/addons.go:174]: Oops my bad here

**Comment by SCedricThomas** [comment]
> The failed CI and the release of go-scalingo are 2 completely different things. I don't understand why there is a link between these two?

The CI build is failing because I locally linked go-scalingo to master

File: cmd/addons.go [hunk 3]
```diff
@@ -160,4 +160,46 @@ var (
 			autocomplete.CmdFlagsAutoComplete(c, "addons-info")
 		},
 	}
+	addonsConfigCommand = cli.Command{
+		Name:     "addons-config",
+		Category: "Addons",
+		Usage:    "Configure an add-on attached to your app",
+		Flags: []cli.Flag{&appFlag,
+			&cli.StringFlag{Name: "maintenance-window-hour", Usage: "Configure add-on maintenance window starting hour"},
```
[the remark this answers is not in the document]
Comment (SCedricThomas) [reply cmd/addons.go:168]: It's the local timezone ! I will update the flag :+1:

File: cmd/addons.go [hunk 4]
```diff
@@ -160,4 +161,54 @@ var (
...
+
+			if c.IsSet("maintenance-window-hour") {
+				config.MaintenanceWindowHour = utils.IntPtr(c.Int("maintenance-window-hour"))
+			}
+
+			if config.MaintenanceWindowHour == nil && config.MaintenanceWindowDay == nil {
+				errorQuitWithHelpMessage(errors.New(ctx, "cannot update your addon without a specified option"), c, "addons-config")
+			}
+
+			err := addons.UpdateAddonConfig(ctx, currentApp, currentAddon, config)
```
Comment (EtienneM) [inline cmd/addons.go:200]: best practice:  I'll be a bit picky on this one, there is a linter which should raise an error here. It's not useful to add the package name in the method name. Hence:
```suggestion
			err := addons.UpdateConfig(ctx, currentApp, currentAddon, config)
```

**Review by EtienneM — APPROVED** [review]
I'm already approving the PR but I would love if you could:
- take into account the remaining comments
- release a new go-scalingo version and use it here

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 ### To be Released
 
+* chore(term): remove `github.com/andrew-d/go-termutil`, use standard library instead
+* feat(cmd): addon can be retrieve from addon type, not only UUID
+* feat(addons): add maintenance windows manipulation with the new `addon-config` command
+
 * chore(term): remove `github.com/andrew-d/go-termutil`, use standard library instead
 * feat(cmd): addon can be retrieve from addon type, not only UUID
```
Comment (curzolapierre) [inline CHANGELOG.md:11]: There are still some duplication
```suggestion

* chore(term): remove `github.com/andrew-d/go-termutil`, use standard library instead
* feat(cmd): addon can be retrieve from addon type, not only UUID
* feat(addons): add maintenance windows manipulation with the new `addon-config` command

```

**Review by yohann-bacha — APPROVED** [review]
great job! but we should be sure that the utc conversions are done

**Pull request merged**